qlib/backtest/__init__.py

```diff
@@ -107,6 +107,24 @@ def get_exchange(
         )
         return exchange
     else:
+        # If user passes an exchange config, inject missing basic kwargs such as freq/start/end.
+        if isinstance(exchange, dict):
+            ex_cfg = copy.deepcopy(exchange)
+            ex_kwargs = ex_cfg.setdefault("kwargs", {})
+            ex_kwargs.setdefault("freq", freq)
+            ex_kwargs.setdefault("start_time", start_time)
+            ex_kwargs.setdefault("end_time", end_time)
+            ex_kwargs.setdefault("codes", codes)
+            if deal_price is not None:
+                ex_kwargs.setdefault("deal_price", deal_price)
+            if subscribe_fields:
+                ex_kwargs.setdefault("subscribe_fields", subscribe_fields)
+            if limit_threshold is not None:
+                ex_kwargs.setdefault("limit_threshold", limit_threshold)
+            ex_kwargs.setdefault("open_cost", open_cost)
+            ex_kwargs.setdefault("close_cost", close_cost)
+            ex_kwargs.setdefault("min_cost", min_cost)
+            exchange = ex_cfg
```

Comment:
<img width="613" height="400" alt="image" src="https://github.com/user-attachments/assets/45b338e4-7622-4bb6-8580-077238e22dee" />


How about this implementation? would it be more elegant?